### DISCUSSION TITLE
How can I allow all literal types?

### DISCUSSION THREAD

**Discussion by nandorojo** [open-circle/valibot#469] [Q&A] [answered]
```ts
type Vendor = 'google' | 's3'

const vendor = v.literal<Vendor>('google')
```

What's the best way here to let me type any (or, better yet, all) of the possible values from `Vendor`? It would be nice to have `literals` plural or something.

The implementation could use an array or something like that? Or multiple args?

**Comment by fabian-hiller** [comment, marked as answer]
I would recommend [`picklist`](https://valibot.dev/api/picklist/). Check it out in our [playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgDUBTVgE2hwAvJgB0YYEwDW5DgAoA2oQDmECMtLdCAGiKoAzIQC6ASgb0YATzDc4PfoJEZRAeQCuMMB4A8VmxGw7XgEoAD5GAHoIuAAVAAtgdETMMmA+ZjZUCE1RUnU5Z1RkbG4ABWQoVG45exDdQgNCEzNI6PjkjtYScnSWdmzuUW4oPCgC0SKS8srq2uh67DUmsyA).

```ts
import * as v from 'valibot';

const Vendor = v.picklist(['google', 's3']);
```
